aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -745,18 +746,19 @@ Tensor scaled_dot_product_attention(
           query_, key, value, attn_mask, compute_logsumexp, dropout_p, is_causal, scale);
       return std::get<0>(out_and_lse);
     }
-    case sdp::SDPBackend::overrideable: {
-      std::optional<Tensor> attn_mask = convert_boolean_attn_mask(attn_mask_, query_.dtype());
+    case SDPBackend::overrideable: {
       auto out_lse_softmax = at::_scaled_dot_product_fused_attention_overrideable(
           query_, key, value, attn_mask, dropout_p, is_causal, false /*return_debug_mask*/, scale);
       return std::get<0>(out_lse_softmax);
     }
-    case sdp::SDPBackend::math: {
-      std::optional<Tensor> attn_mask = convert_boolean_attn_mask(attn_mask_, query_.dtype());
-      if ((!GradMode::is_enabled() || (!query_.requires_grad() && !key.requires_grad() && !value.requires_grad()))
-          && query_.device().type() == DeviceType::MPS && dropout_p == 0.0
-          && query_.is_contiguous() && key.is_contiguous() && value.is_contiguous()
-          && !query_.is_nested() && !key.is_nested() && !value.is_nested()) {
+    case SDPBackend::math: {
+      const auto any_nested = query_.is_nested() || key.is_nested() || value.is_nested();
```

Comment:
So these checks should ideally be in the dispatching logic, I am not sure if MPS currently has a stub for doing the input validation. I think that it this length of check its okay but if it gets any larger I think we should do it properly